[itssisi] IP

itssisi opened the pull request:
# Snow ☃️
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Snow is here to help you remember the things you need to do. No need to worry about forgetting any tasks! 
Snow is,
- text-based
- user-friendly
- ~~SLOW~~ VERY FAST 🤩

All you need to is, 
1. download it from 
2. double-clicked
3. add your tasks
4. let it manage your tasks for you 🥳

And it is ~~NOT~~ **FREE**!

Features: 
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon!)

Stay tuned for more cool features!

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

itssisi committed:
Level-0

itssisi committed:
Level 1.Echo

itssisi committed:
Level 2. Add, List

itssisi committed:
Level 3. Mark as Done

itssisi committed:
Level 4. ToDos, Events, Deadlines

itssisi committed:
Level 5. Handle Errors

itssisi committed:
Level 6. Delete

itssisi committed:
Automated Text UI Testing

itssisi committed:
Add create file and write to file feature.

itssisi committed:
Add LocalDate format to deadline.

itssisi committed:
Add OOP

itssisi committed:
Add duke package

itssisi committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2324S1/ip

itssisi committed:
Automate project build

itssisi committed:
Add find

itssisi committed:
Add Javadoc comments

itssisi committed:
Tweak the code to comply with a coding standard

itssisi committed:
Merge branch 'branch-A-JavaDoc'

itssisi committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java
#	src/main/java/duke/Todo.java

itssisi committed:
Use CheckStyle

itssisi committed:
Add a GUI implemented by JavaFX technology to the chatbot

Originalidk commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,44 @@
+            String[] splitOutput = userOutput.split(" ");
+            if (userOutput.equals("list")) {
+                inputList.list();
+            } else if (splitOutput[0].equals("deadline")) {
+                Deadline.setDeadline(userOutput, inputList);
+            } else if (splitOutput[0].equals("todo")) {
+                Todo.setTodo(userOutput, inputList);
+            } else if (splitOutput[0].equals("event")) {
+                Event.setEvent(userOutput, inputList);
+            } else if (splitOutput[0].equals("delete")) {
+                inputList.delete(splitOutput);
+            } else if (splitOutput[0].equals("find")) {
+                inputList.find(splitOutput);
+            } else if (splitOutput[0].equals("mark")) {
+                inputList.mark(splitOutput);
+            } else if (splitOutput[0].equals("unmark")) {
```
Perhaps you could assign splitOutput[0] to a variable name

Originalidk commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,29 @@
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public static void horizontalLine() {
+        System.out.println("____________________________________________________________");
+    }
```
Perhaps having this as a constant instead of a function would work better? Also, I could not find this being used elsewhere.

raydenlim replied:
@Originalidk its used on line 12 and 15 in Ui.java haha, 
Maybe you could change the naming to printHorizontalLine(), so that it is clearer to users what the command does :)
i.e print a horizontal line

Originalidk commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+     */
+    public static void setDeadline(String userOutput, TaskList inputList) throws Exception {
+        try {
+            String newDes = userOutput.split("deadline")[1].split("/by")[0].strip();
```
Not sure if the name newDes is clear.

Originalidk commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+        String userOutput = userInput.nextLine(); // Read user input
+        while (!userOutput.equals("bye")) {
+            Boolean isBye = Parser.command(userOutput, inputList);
+            if (isBye) {
+                break;
+            }
```
Not sure if the break is necessary since your while loop should end once the user types in "bye".

Originalidk commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,52 @@
+            }
+            Todo newTodo = new Todo(todoCommand);
+            inputList.add(newTodo);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(newTodo);
+            System.out.println("Now you have " + inputList.size() + " tasks in the list.");
```
Perhaps you could create a separate method to print the statements. Noticed this elsewhere too.

Originalidk reviewed (changes requested):
LGTM. Just a few nits I noticed.

raydenlim commented on src/main/java/duke/Ui.java:
Solid work! Keep it up :)
Perhaps you might want to consolidate all the System.out.println(...) to specific methods in the Ui file, and update the corresponding print methods in each file with the methods. This will make the code cleaner and improve clarity.

This will also make the code easier to maintain in the long run, as you only need to edit 1 file, the ui.java file, instead of going through all the files (e.g task, event, deadline etc) to update the print statements.

raydenlim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    public void run() {
+        ui.identity();
```
the name identity() is a little ambiguous for me. 
> Use nouns for things and verbs for actions

Perhaps showIdentity()?

raydenlim commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,44 @@
+/**
+ * Parser splits the user command to be processed by the bot.
+ */
+public class Parser {
```
Wow, diddnt think about using parser as a boolean. This is so cool and neat!

raydenlim commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,28 @@
+package duke;
+
+public abstract class Task {
```
I think its abit weird to call this an abstract class when theres no abstract method.
Maybe my understanding is wrong haha, I'm sorry in advance

itssisi committed:
Use assertion of add method in TaskList

TaskList should not be empty after a task is being added into the list.

Assert is ensuring TaskList is not empty.

itssisi committed:
Improves code quality

There are some methods without Javadocs comments.

Adding comments to explain the methods improves code quality and understandability.

itssisi committed:
Merge pull request #2 from itssisi/branch-A-CodeQuality

Improves code quality

itssisi committed:
Merge pull request #1 from itssisi/branch-A-Assertions

Use assertion of add method in TaskList

itssisi committed:
Updates task easily using update() method

There is no update method in the task, so task has to be
deleted before being updated.

Updating tasks when update command is invoked by user,
without deleting the task first.

itssisi committed:
Update README.md

itssisi committed:
Update README.md

itssisi committed:
Add a product website

itssisi committed:
Update README.md
